### PR TITLE
Fix empty slugs for questions after update from 1.0.5 to 1.0.6

### PR DESCRIPTION
**Pull request by cyberstar**
Fixes empty slugs (were added in 1.0.6 version) by performing same actions as in ``aldryn-translation-tools.TranslatedAutoSlugifyMixin`` but without flexibility since all parameters are known in advance.

Previous approach of invoking save method of ``Question`` model from real models (not orm) is not reliable since any new changes to the ``Question`` model would break it.

 - [x] Django 1.6 (south) data mogration
 - [x] Django 1.7 data migration

This migrations were tested as follows:

* install ``aldryn-faq`` 1.0.5
* create feq questions
* update ``aldryn-faq`` to latest (1.0.6) + migrations
* modify models and add an int field, create new migration
* run migrations (the new data migrations + new int field migration)
* ensure migrations are finished and add-on is working
* ensure that without int field migrations are working.

### CODE REVIEW

**Comment by mkoistinen** [comment]
LGTM.

File: aldryn_faq/migrations/0008_generate_slugs_for_existing_questions.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    """
+    Generate the slug candidate the same way as
+    aldryn-translation-tools.TranslatedAutoSlugifyMixin does that.
+    Simplified, since we know in advance
+    """
+    if not getattr(translation, 'slug', ''):
+        slug_source = translation.title
+        slug = force_text(slugify(slug_source))
+        candidate = force_text(slugify(slug_source))
+        qs = question_model.objects.filter(translations__language_code=language_code).exclude(pk=translation.master_id)
```
Comment (intellectronica) [inline aldryn_faq/migrations/0008_generate_slugs_for_existing_questions.py]: This line is so looooooooooooong.

Please keep to 80 columns.

File: aldryn_faq/south_migrations/0023_generate_slugs_for_existing_questions.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    """
+    Generate the slug candidate the same way as
+    aldryn-translation-tools.TranslatedAutoSlugifyMixin does that.
+    Simplified, since we know in advance
+    """
+    if not getattr(translation, 'slug', ''):
+        slug_source = translation.title
+        slug = force_text(slugify(slug_source))
+        candidate = force_text(slugify(slug_source))
+        qs = question_model.objects.filter(translations__language_code=language_code).exclude(pk=translation.master_id)
```
Comment (intellectronica) [inline aldryn_faq/south_migrations/0023_generate_slugs_for_existing_questions.py]: Long

**Comment by intellectronica** [comment]
:+1:

**Comment by mkoistinen** [comment]
So, I'm not crazy about all the duplicate code in the migration file. I think I would have instead called a utility function that both the model and migration can call as required. That's going to get out of sync real fast. As in not DRY. =(

**Comment by cyberstar** [comment]
Depends on unreleased ``aldryn-translation-tools`` version (0.1.3). without it tests would fail, since migration are now depend on  slug generation logic from it.

**Comment by Chive** [comment]
In general, code imports in migrations are a bad idea. If you're doing that you basically say that aldryn-translation-tools will be a requirement forever for this package and ``TranslatedAutoSlugifyMixin`` may never be removed / changed.

**Comment by cyberstar** [comment]
rebased, setup in master already bumps ``aldryn-translation-tools`` to 0.2.1

**Comment by mkoistinen** [comment]
Nicely done!

**Pull request merged**